pyart/correct/despeckle.py

```diff
@@ -0,0 +1,474 @@
+                                 np.sin(np.deg2rad(delta))):
+                return False
+            else:
+                return True
+        else:
+            return False
+    else:
+        return False
+
+
+def _check_sweeps(sweeps, radar):
+    """
+    Parse the sweeps keyword and convert it to a list of ints.
+    The output will be iterated over.
+
+    Parameters
+    ----------
+    sweeps : int or list of ints or None
+        Sweep numbers to put into an iterable list. If None, all sweeps in the
+        radar object will be examined.
+    radar : pyart.core.Radar object
+        Radar object to query.
+
+    Returns
+    -------
+    sweeps : list of ints
+        Sweep numbers as an iterable list
+
+    """
+    if sweeps is None:
+        sweeps = np.arange(len(radar.sweep_number['data']))
+    else:
+        if hasattr(sweeps, '__len__'):
+            sweeps = np.asarray(sweeps)
+        else:
+            sweeps = np.asarray([sweeps])
+    return sweeps
+
+
+def _check_threshold(threshold):
+    """
+    Parse the threshold keyword and return the lower and upper boundaries for
+    the object search.
+
+    Parameters
+    ----------
+    threshold : int or float, or 2-element tuple of ints or floats
+        Threshold values above (if single value) or between (if tuple)
+        for objects to be identified.
+
+    Returns
+    -------
+    tlo : int or float
+        Lower bound for the threshold. Values below this will not be included
+        in the hunt for objects.
+    thi : int or float or None
+        Upper bound for the threshold. Values above this will not be included
+        in the hunt for objects. None means no upper bound.
+
+    """
+    if not hasattr(threshold, '__len__'):
+        threshold = np.asarray([threshold])
+    if len(threshold) == 2:
+        tlo = threshold[0]
+        thi = threshold[1]
+    elif len(threshold) > 2 or np.ndim(threshold) > 1:
+        raise IndexError('Fix threshold argument! Must be single scalar ' +
+                         'or 2-element tuple')
+    else:
+        tlo = threshold[0]
+        thi = None
+    return tlo, thi
+
+
+def _generate_dict(label_storage):
+    """
+    Build the dictionary that includes all the object label information.
+    If the entire Radar object was searched, the dictionary is ready to
+    be added as a new field.
+
+    Parameters
+    ----------
+    label_storage : 2D array of ints
+        Object labels as a 2D array
+
+    Returns
+    -------
+    label_dict : dict
+        Dictionary containing object labels and associated metadata
+
+    """
+    label_dict = {}
+    label_dict['data'] = label_storage
+    label_dict['units'] = 'None'
+    label_dict['long_name'] = 'Objects in Scan'
+    label_dict['standard_name'] = 'objects_in_scan'
+    label_dict['coordinates'] = 'elevation azimuth range'
+    label_dict['valid_max'] = np.max(label_storage)
+    label_dict['valid_min'] = 1
```

Comment:
A better location for this metadata would be the configuration file but I'm not against keeping it here for simplicity